### ISSUE TITLE
Support MongoDB

### ISSUE THREAD

**Issue by JasonTypesCodes** [jhipster/generator-jhipster-micronaut#49]

**Comment by amanganiello90** [comment]
Hi!! I would like to do this task. 
So I have not found an ORM for mongodb. Is it necessary to use [mongodb-driver-sync](https://micronaut-projects.github.io/micronaut-mongodb/latest/guide/) that uses mongoClient?
A minimal configurated [example](https://dzone.com/articles/micronaut-mongodb).

**Comment by atomfrede** [comment]
Yes I think there is currently no support in e.g. micronaut data like in spring boot data.

**Comment by amanganiello90** [comment]
ookkk, I start with that. Thanks

**Comment by atomfrede** [comment]
Maybe [gorm for mongodb ](http://gorm.grails.org/latest/mongodb/manual/)is something to look at. https://micronaut-projects.github.io/micronaut-mongodb/latest/guide/

**Comment by amanganiello90** [comment]
> 
> 
> Maybe [gorm for mongodb ](http://gorm.grails.org/latest/mongodb/manual/)is something to look at. https://micronaut-projects.github.io/micronaut-mongodb/latest/guide/

>
> For Groovy users and users familiar with Grails, special support has been added to Micronaut for using GORM for MongoDB.

It seems that support only groovy language and not java.

**Comment by atomfrede** [comment]
Ah you might be right, was not clear to me from the single sentence.

**Comment by JasonTypesCodes** [comment]
@atomfrede is correct that Gorm requires Groovy.  While it wouldn't be a requirement to add support for Mongo here, we do have it on our upcoming roadmap in Micronaut: https://github.com/micronaut-projects/micronaut-core/wiki/Top-Level-Roadmap

**Comment by amanganiello90** [comment]
> 
> 
> @atomfrede is correct that Gorm requires Groovy. While it wouldn't be a requirement to add support for Mongo here, we do have it on our upcoming roadmap in Micronaut: https://github.com/micronaut-projects/micronaut-core/wiki/Top-Level-Roadmap

So I can start the issue implementing the mongodb-driver or we should wait for the groovy support?

**Comment by atomfrede** [comment]
Two things to keep in mind:

* Doing it without proper micronaut data support would involve manual mapping from mongo document to entity/business. This can be done, but its a lot of work we would remote again when micronaut data supports it
* Adding large features now might also not the best idea with the first jhipster 7 beta/pre-release around the corner

**Comment by amanganiello90** [comment]
Ok, then I will wait for the jhipster/mongodb supporting and upgrade. Thanks

**Comment by JasonTypesCodes** [comment]
I wanted to leave a note here to say that while Micronaut Data does not support Mongo yet (https://github.com/micronaut-projects/micronaut-data/issues/220), there is a Micronaut MongoDB module here: https://micronaut-projects.github.io/micronaut-mongodb/latest/guide/

**Comment by dstepanov** [comment]
FYI Micronaut Data supports MongoDB already
